core/GlobalState.cc

```diff
@@ -1771,6 +1771,18 @@ void GlobalState::deleteMethodSymbol(MethodRef what) {
     this->methods[what.id()] = this->methods[0].deepCopy(*this);
 }
 
+void GlobalState::deleteFieldSymbol(FieldRef what) {
+    ENFORCE(what.data(*this)->flags.isField);
```

Comment:
:cloud_yell:

If you want, the WIP changes I have to support static-fields on fast path have mostly been minor. We may be able to sneak in a refactor to split things into FieldRef and StaticFieldRef after this PR and before I land the static-fields change.